### PR TITLE
Multicellular Freebuild

### PR DESCRIPTION
**Pull request by Nabstablues**
**Brief Description of What This PR Does**

This PR adds a basic freebuild editor for the early multicellular.

**Related Issues**

N/A

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [ ] Initial code review passed (this and further items should not be checked by the PR author)
- [ ] Functionality is confirmed working by another person (see above checklist link)
- [ ] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author. Merging must follow our
[styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md#git).

### CODE REVIEW

File: src/general/GameProperties.cs [hunk 1]
```diff
@@ -110,6 +110,13 @@ public static GameProperties StartNewEarlyMulticellularGame(WorldGenerationSetti
 
         game.EnterPrototypes();
 
+        if (freebuild)
+        {
+            game.EnterFreeBuild();
+            game.GameWorld.GenerateRandomSpeciesForFreeBuild();
```
Comment (hhyyrylainen) [inline src/general/GameProperties.cs:116]: Thanks for contributing to Thrive. Someone will do a full review once the diff for MainMenu.cs is fixed (it seems you converted all spaces to tabs), but I'll just quickly comment that could you place a todo comment here like: `// TODO: generate some other multicellular species for the player to interact with`.

**Comment by Kashnox** [comment]
I disabled whitespace checking so I could take a look at MainMenu.cs and everything looks good there, apart from the formatting. The only other comment I have is that we should probably rename the old freebuild button and its related methods and fields to include the word "microbe" to match the new multicellular freebuild stuff.

This will be a really helpful feature, by the way. I've been doing a lot of work with the multicellular editor and this will make it much easier to test.

**Comment by hhyyrylainen** [comment]
I'll just point out that the fastest way to test the multicellular editor is to use `res://src/stage_starters/EarlyMulticellularEditorStarter.tscn` as the startup scene. That's even faster than what you'd get by having to interact with a multicellular freebuild button.

**Comment by Nabstablues** [comment]
Yes, but in freebuild you don't have to worry about mp limitations and getting recourses, and this also allows someone to get into the multicellular editor without messing with any code.

**Comment by hhyyrylainen** [comment]
Of course, I wasn't rebuking the usefulness of this additional feature. Just saying that for purely debugging purposes there's those stage starter scenes I've added for easily debugging any stage / editor, which I personally use all the time.

**Comment by ender-titan1** [comment]
This seems to have been abandoned, I could try to finish this if the original author won't do anything with this.

**Comment by hhyyrylainen** [comment]
I think that would be fine if @Nabstablues doesn't reply in a day or so. It'd be great to finish this PR to get this pretty simple change completed and off the list of open PRs.

**Comment by ender-titan1** [comment]
This should be all good now, other than the BOM, which I can't figure out for the life of me how to manually add to a file.

**Comment by hhyyrylainen** [comment]
Depends on your editor but it is usually close to the file encoding settings. Rider has a separate button in the encoding menu to add BOM. Emacs can save a file with a BOM when using an utf8 encoding variant with signature in the name.

**Review by hhyyrylainen — APPROVED** [review]
I fixed the MainMenu.cs file by checking it out from master and only letting the changes from this PR through that were intended (I removed the various comment splitting on multiple lines).

Also I fixed keyboard navigation for the new button as it had incorrectly copied settings from the above button.

**Pull request merged**